native/acter/src/api/verification.rs

```diff
@@ -114,10 +114,65 @@ impl VerificationEvent {
         self.emojis.clone_from(&emojis);
     }
 
-    pub fn get_emojis(&self) -> Vec<VerificationEmoji> {
+    // when this device triggered verification of other device, it can get emojis from SAS state
+    pub fn emojis(&self) -> Vec<VerificationEmoji> {
         self.emojis.clone()
     }
 
+    // when other device triggered verification of this device, it can get emojis from remote server
+    pub async fn get_emojis(&self) -> Result<Vec<VerificationEmoji>> {
+        let client = self.client.clone();
+        let sender = self.sender.clone();
+        let event_id = self.event_id.clone();
+        let txn_id = self.txn_id.clone();
+        RUNTIME
+            .spawn(async move {
+                if let Some(event_id) = event_id {
+                    if let Some(Verification::SasV1(sas)) = client
+                        .encryption()
+                        .get_verification(&sender, event_id.as_str())
+                        .await
+                    {
+                        if let Some(items) = sas.emoji() {
+                            let sequence = items
+                                .iter()
+                                .map(|e| VerificationEmoji {
+                                    symbol: e.symbol.chars().next().unwrap() as u32, // first char in string
+                                    description: e.description.to_string(),
+                                })
+                                .collect::<Vec<VerificationEmoji>>();
+                            return Ok(sequence);
+                        } else {
+                            return Ok(vec![]);
+                        }
+                    }
+                } else if let Some(txn_id) = txn_id {
+                    if let Some(Verification::SasV1(sas)) = client
+                        .encryption()
+                        .get_verification(&sender, txn_id.as_str())
+                        .await
+                    {
+                        if let Some(items) = sas.emoji() {
+                            let sequence = items
+                                .iter()
+                                .map(|e| VerificationEmoji {
+                                    symbol: e.symbol.chars().next().unwrap() as u32, // first char in string
+                                    description: e.description.to_string(),
+                                })
+                                .collect::<Vec<VerificationEmoji>>();
+                            return Ok(sequence);
+                        } else {
+                            return Ok(vec![]);
+                        }
+                    }
+                }
+                // request may be timed out
+                info!("Could not get verification object");
+                Ok(vec![])
```

Comment:
And here. This sounds like an error case to me...
```suggestion
                bail!("Could not get verification object");
```

Reply:
Other device may not respond in time and timeout is possible case.
If this function calls `bail!()`, flutter may occur exception.
I wonder if flutter should occur exception in possible case.

Reply:
Yes, flutter should _not_ consider any case where we don't see the emoji a regular case of operation. Raising/throwing here is the right thing to do.

Reply:
Understood

Reply:
Should I replace `Ok(false)` with `bail!()` here too?
https://github.com/bitfriend/a3/blob/d3b14ed0add97f31c0c55f897925f668dcccce6b/native/acter/src/api/verification.rs#L204-L206

Reply:
yes, probably.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/886/commits/1ae92a8818f9db6af242ee3820ac9a45bfa7102a